ocaml/xapi/xapi_pif.ml

```diff
@@ -767,6 +767,31 @@ let set_property ~__context ~self ~name ~value =
         Nm.bring_pif_up ~__context pif
     ) (self :: vlan_pifs)
 
+let pif_has_clusters ~__context (self : API.ref_PIF) =
+  debug "Looking up clusters for PIF: %s" (Ref.string_of self);
+  let network_of_pif =
+    try Db.PIF.get_network ~__context ~self with _ ->
+      failwith (Printf.sprintf "Could not find network of PIF: %s" (Ref.string_of self))
+  in
+  info "Got network: %s" (Ref.string_of network_of_pif);
```

Comment:
I would remove the logging in this function. We are logging far too much already...